trieste/acquisition/function.py

```diff
@@ -509,6 +512,155 @@ def prepare_acquisition_function(
         return lambda at: expected_improvement(objective_model, eta)(at) * constraint_fn(at)
 
 
+class ExpectedHypervolumeImprovement(SingleModelAcquisitionBuilder):
+    """
+    Builder for the :func:`expected_hv_improvement` acquisition function.
+    The implementation of the acquisition function largely
+    follows :cite:`yang2019efficient`
+    """
+
+    def __repr__(self) -> str:
+        """"""
+        return "ExpectedHypervolumeImprovement()"
+
+    def prepare_acquisition_function(
+        self, dataset: Dataset, model: ProbabilisticModel
+    ) -> AcquisitionFunction:
+        """
+        :param dataset: The data from the observer. Must be populated.
+        :param model: The model over the specified ``dataset``.
+        :return: The expected_hv_improvement function.
+        """
+        tf.debugging.assert_positive(len(dataset), message="Dataset must be populated.")
+        mean, _ = model.predict(dataset.query_points)
+
+        _pf = Pareto(mean)
+        _reference_pt = get_reference_point(_pf.front)
```

Comment:
If you are already passing _pf to expected_hv_improvement, why do you also need to pass _reference_pt ?

Reply:
The reason is more for a future extension consideration: it might also be good to allow user specify his self defined `get_reference_point`. As for the reference point setting, to my best of knowledge there is no unique agreed solution. Hence, in future if a user has specified his own `modified_get_reference_point`, it might be good to proceed the reference point at the prepare stage and feed into the acquisition function for calculation & optimization.

Reply:
indeed choosing the reference point is an open question and this flexibility is important

Reply:
Yep that makes sense!